lib/src/components/NxCombobox/__tests__/NxCombobox.test.tsx

```diff
@@ -0,0 +1,522 @@
+        dropdownElement = getByRole('listbox', { hidden: true });
+
+    expect(dropdownElement).toBeInTheDocument();
+  });
+
+  it('sets an id on the dropdown and references it in the combobox input aria-controls', function() {
+    const { getByRole } = quickRender({ matches: [{ id: '1', displayName: 'Foo' }] }),
+        inputElement = getByRole('combobox'),
+        dropdownElement = getByRole('listbox', { hidden: true });
+
+    expect(dropdownElement).toHaveAttribute('id');
+    expect(inputElement).toHaveAttribute('aria-controls', dropdownElement.id);
+  });
+
+  it('sets the alert role on the alert dropdown when it is in loading, error, or empty states', function() {
+    const { getAllByRole, rerender } = quickRender({ value: 'f' }),
+        alertDropdownElement = getAllByRole('alert', { hidden: true })[1];
+
+    expect(alertDropdownElement).toBeInTheDocument();
+
+    rerender(<NxCombobox { ...minimalProps } loading={true} />);
+    expect(alertDropdownElement).toBeInTheDocument();
+
+    rerender(<NxCombobox { ...minimalProps } loadError={'boo'} />);
+    expect(alertDropdownElement).toBeInTheDocument();
+  });
+
+  it('sets aria-live on the alert dropdown to "polite"', function() {
+    expect(quickRender().getAllByRole('alert', { hidden: true })[1]).toHaveAttribute('aria-live', 'polite');
+  });
+
+  it('sets aria-busy on the alert dropdown if loading is true', function() {
+    const { getAllByRole, rerender } = quickRender(),
+        alertDropdownElement = getAllByRole('alert', { hidden: true })[1];
+
+    expect(alertDropdownElement).toHaveAttribute('aria-busy', 'false');
+
+    rerender(<NxCombobox { ...minimalProps } loading={true} />);
+    expect(alertDropdownElement).toHaveAttribute('aria-busy', 'true');
+  });
+
+  it('renders error and textcontent when the `loadError` prop is set', function() {
+    const alertDropdownElement = quickRender({ loadError: 'err' }).getAllByRole('alert', { hidden: true })[1];
+
+    expect(alertDropdownElement).toBeInTheDocument();
+    expect(alertDropdownElement).toHaveTextContent('err');
+  });
+
+  it('fires onSearch with the search text when the retry button is clicked', async function() {
+    const onSearch = jest.fn(),
+        { getByRole} = quickRender({ loadError: 'err', onSearch }),
+        inputElement = getByRole('combobox');
+
+    expect(onSearch).not.toHaveBeenCalled();
+    await userEvent.type(inputElement, 'f');
+    const retryBtn = getByRole('button', { name: /retry/i });
+    await userEvent.click(retryBtn);
+    expect(onSearch).toHaveBeenCalledWith('f');
+  });
+
+  it('fires onChange when the button with role option is clicked', async function() {
+    const onChange = jest.fn(),
+        { getByRole, getAllByRole } = quickRender({
+          matches: [{ id: '1', displayName: 'Foo' }, { id: '2', displayName: 'Boo' }], onChange }),
+        inputElement = getByRole('combobox');
+
+    await userEvent.type(inputElement, 'o');
```

Comment:
With the current versions of `userEvent`, you're actually supposed to use `userEvent.setup()` in each test.  See https://testing-library.com/docs/user-event/setup

Reply:
Thanks for the info, updated here 6dbc601488edec3acdf06ca1c289688de300a1a8.